packages/skygear-core/src/types.ts

```diff
@@ -53,7 +53,20 @@ export interface User {
 /**
  * @public
  */
-export type Identity = PasswordIdentity | OAuthIdentity | CustomTokenIdentity;
+export interface Identity {
+  type: string;
+  claims: IdentityClaims;
+}
+
+/**
+ * @public
+ */
+export interface IdentityClaims {
+  [claim: string]: unknown;
+  email?: string;
+  phone?: string;
+  username?: string;
```

Comment:
Just keep `email` first? Btw `phone` in OIDC standard claims is called `phone_number` actually.